cmd/run.go

```diff
@@ -60,11 +61,27 @@ func Run(cmd *cobra.Command, args []string) {
 			cmd.Help()
 			os.Exit(1)
 		}
-		repos, err := github.GetRepositories(username)
+
+		provider, err := cmd.Flags().GetString("provider")
 		if err != nil {
-			l.Println(err)
+			panic(err)
+		}
+
+		repos := make([]string, 0)
+
+		switch provider {
+		case "github":
+			repos, err = github.GetRepositories(username)
+			if err != nil {
+				l.Println(err)
+			}
+			break
+		case "gitlab":
+			repos, err = gitlab.GetRepositories(username)
```

Comment:
need to log the error for gitlab as well. You try to put this validation outside of the switch to be used for both cases.